corehq/apps/userreports/sql/adapter.py

```diff
@@ -48,20 +48,26 @@ def save(self, doc):
         indicator_rows = self.config.get_all_values(doc)
         if indicator_rows:
             table = self.get_table()
-            with self.engine.begin() as connection:
-                # delete all existing rows for this doc to ensure we aren't left with stale data
-                delete = table.delete(table.c.doc_id == doc['_id'])
-                connection.execute(delete)
-                for indicator_row in indicator_rows:
-                    all_values = {i.column.database_column_name: i.value for i in indicator_row}
-                    insert = table.insert().values(**all_values)
-                    try:
-                        connection.execute(insert)
-                    except IntegrityError:
-                        # Someone beat us to it. Concurrent inserts can happen
-                        # when a doc is processed by the celery rebuild task
-                        # at the same time as the pillow.
-                        pass
+            try:
+                with self.engine.begin() as connection:
+                    # delete all existing rows for this doc to ensure we aren't left with stale data
+                    delete = table.delete(table.c.doc_id == doc['_id'])
+                    connection.execute(delete)
+                    bad_saves = False
+                    for indicator_row in indicator_rows:
+                        all_values = {i.column.database_column_name: i.value for i in indicator_row}
+                        insert = table.insert().values(**all_values)
+                        try:
+                            connection.execute(insert)
+                        except (IntegrityError, InternalError):
+                            bad_saves = True
+                    if bad_saves:
+                        raise BadSaveError()
```

Comment:
The "current transaction is aborted, commands ignored until end of transaction block" error is what happens when a query is executed after a bad query is executed and then another query is attempted before the transaction is rolled back. So the bug in the original code was the `try`/`except` inside the `for`-loop. Updated the code block above with a less complicated solution.